src/modules/sdlog2/sdlog2.c

```diff
@@ -1280,8 +1313,67 @@ int sdlog2_thread_main(int argc, char *argv[])
 	/* running, report */
 	thread_running = true;
 
+	// wakeup source
+	px4_pollfd_struct_t fds[1];
+
+	int poll_counter = 0;
+
+	int poll_to_logging_factor = 1;
+
+	if (record_replay_log) {
+		subs.replay_sub = orb_subscribe(ORB_ID(ekf2_replay));
+		fds[0].fd = subs.replay_sub;
+		fds[0].events = POLLIN;
+		poll_to_logging_factor = 1;
+	} else {
+		subs.sensor_sub = orb_subscribe(ORB_ID(sensor_combined));
```

Comment:
I wasn't sure if we then know 100% that we log all replay messages. Couldn't it be that the app misses a sample then?